Михаил установил на свой телефон обновление и хочет проверить, увеличилось ли среднее время работы телефона от одного заряда. Для этого он рассмотрел значения длительности работы от одного заряда до и после обновления. Считая, что время работы телефона распределено нормально, а дисперсия одинакова в обоих случаях, проверить на уровне значимости 0.05 гипотезу о том, что среднее время работы не изменилось.
Введите модуль функции отклонения Po(X):

Решение:
Н_0 -  среднее время работы не изменилось
Н_1 среднее время работы изменилось
Найдем |func_deviation| и если он < С, то Гипотезу Н_0 принимаем, если модуль функции отклонения > С, следовательно Н_0 - отвергаем 

In [9]:
import pandas as pd
import numpy as np
import scipy.stats as st
from scipy.stats import t

In [4]:
d_before = pd.read_csv('/Users/ekaterina/github_repos/ITMO_STUDY/Zadachi_ITMO_teorver_statistica/hypothesis_5_X.csv', header=None)
d_before

0
0  35.857419
1  29.851216
2  25.991484
3  34.426247
4  33.322775
5  36.441769
6  27.858808
7  31.017693
8  26.438346
9  30.520649

In [6]:
d_after = pd.read_csv('/Users/ekaterina/github_repos/ITMO_STUDY/Zadachi_ITMO_teorver_statistica/hypothesis_5_Y.csv', header=None)
d_after

0
0  38.829230
1  32.936778
2  39.731949
3  37.074383
4  33.338638
5  35.652129
6  33.100469

In [ ]:
n=10  # Объем выблорки Х
m=7  # Объем выблорки Y
e=0.05

In [25]:
print("Mean_X", d_before.mean())
print("Mean_Y", d_after.mean())

Mean_X 0    31.172641
dtype: float64
Mean_Y 0    35.809082
dtype: float64


In [28]:
S_X = (np.var(d_before[0], ddof=1))
print("Дисперсия несмещенная S_X", S_X)

S_Y = (np.var(d_after[0], ddof=1))
print("Дисперсия несмещенная S_Y", S_Y)

Дисперсия несмещенная S_X 14.150508045553646
Дисперсия несмещенная S_Y 7.970935791994166


In [29]:
print("C", t.ppf(1-(e/2), 15))

C 2.131449545559323


In [40]:
func_deviation = (((n*m)/(n+m))**0.5)*(((d_before.mean()-d_after.mean())/(((n-1)*S_X+(m-1)*S_Y)/(n+m-2))**0.5))

In [43]:

print("If |func_deviation| < c Гипотезу Н_0 принимаем, если > Н_0 - отвергаем ", func_deviation)

If func_deviation < c Гипотезу принимаем, если > - отвергаем  0   -2.75304
dtype: float64


Модуль ф-ии отклонения больше С: 2.75304 > 2.131 следовательно, Гипотезу Н_0 о том, что среднее время работы не изменилось отвергаем.